src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5953,3 +5953,49 @@ void MacroAssembler::double_move(VMRegPair src, VMRegPair dst, Register tmp) {
       strd(src.first()->as_FloatRegister(), Address(sp, reg2offset_out(dst.first())));
   }
 }
+
+// Attempt to fast-lock an object. Fall-through on success, branch to slow label
+// on failure.
+// Registers:
+//  - obj: the object to be locked
+//  - hdr: the header, already loaded from obj, will be destroyed
+//  - t1, t2, t3: temporary registers, will be destroyed
+void MacroAssembler::fast_lock(Register obj, Register hdr, Register t1, Register t2, Register t3, Label& slow) {
+  // Check if we would have space on lock-stack for the object.
+  ldr(t1, Address(rthread, Thread::lock_stack_current_offset()));
+  ldr(t2, Address(rthread, Thread::lock_stack_limit_offset()));
+  cmp(t1, t2);
+  br(Assembler::GE, slow);
```

Comment:
```suggestion
  br(Assembler::HS, slow);
```
Are these really 64-bit unsigned quantities, though?